zerver/lib/create_user.py

```diff
@@ -214,6 +214,13 @@ def create_user(
         force_date_joined=force_date_joined,
         email_address_visibility=user_email_address_visibility,
     )
+
+    if bot_type is not None and bot_owner is not None:
+        if bot_owner.role == UserProfile.ROLE_REALM_ADMINISTRATOR:
+            user_profile.role = UserProfile.ROLE_REALM_ADMINISTRATOR
```

Comment:
Where did you see a request that bots gain the administrator role in the issue description?